### ISSUE TITLE
Indent definition rule not found.

### ISSUE THREAD

**Issue by nwhite89** [AtomLinter/linter-eslint#202]
So I'm looking at moving to `ESLint` from `JSHint` & `JSCS`, I'm getting the following error whilst using the `eslint-linter` for Atom and I'm getting an error stating that `indent` isn't an option?

Any help on this would be greatly appreciated.

Error: 
```
Definition for rule 'indent' was not found.
/Users/nickwhite/.atom/packages/linter-eslint/lib/linter-eslint.coffee:150 Error: Definition for rule 'indent' was not found.
  at /Users/nickwhite/Sites/static_html/node_modules/eslint/lib/eslint.js:591:27
  at Array.forEach (native)
  at EventEmitter.module.exports.api.verify (/Users/nickwhite/Sites/static_html/node_modules/eslint/lib/eslint.js:567:16)
  at /Users/nickwhite/Dropbox/Atom/packages/linter-eslint/lib/linter-eslint.coffee:111:18
  at exports.allowUnsafeNewFunction (/Users/nickwhite/Dropbox/Atom/packages/linter-eslint/node_modules/loophole/lib/loophole.js:25:14)
  at Object.provider.lint (/Users/nickwhite/Dropbox/Atom/packages/linter-eslint/lib/linter-eslint.coffee:109:13)
  at Promise.then._this.emitter.emit.linter (/Users/nickwhite/Dropbox/Atom/packages/linter/lib/linter-registry.coffee:57:22)
  at LinterRegistry.triggerLinter (/Users/nickwhite/Dropbox/Atom/packages/linter/lib/linter-registry.coffee:56:16)
  at /Users/nickwhite/Dropbox/Atom/packages/linter/lib/linter-registry.coffee:50:17
  at Array.map (native)
  at /Users/nickwhite/Dropbox/Atom/packages/linter/lib/linter-registry.coffee:48:27`
```

My config file looks like:
```
{
  "env": {
    "browser": true
  },
  "globals": {},
  "rules": {
    "no-debugger": 0,
    "no-eval": 0,
    "dot-notation": 0,
    "eqeqeq": 0,
    "indent": [2, 2, {
        "VariableDeclarator": 2,
        "SwitchCase": 2
    }],
    "camelcase": [
      2,
      {
        "properties": "never"
      }
    ],
    "curly": [
      2,
      "all"
    ],
    "space-return-throw-case": 0,
    "space-after-keywords": [
      2,
      "always"
    ],
    "space-before-blocks": [
      2,
      "always"
    ],
    "space-before-function-paren": [
      2,
      "always"
    ],
    "one-var": [
      2,
      "never"
    ],
    "no-empty": 2,
    "array-bracket-spacing": [
      2,
      "always",
      {
        "singleValue": true
      }
    ],
    "space-in-parens": [
      2,
      "never"
    ],
    "key-spacing": [
      2,
      {
        "beforeColon": false,
        "afterColon": true
      }
    ],
    "comma-style": [
      2,
      "first"
    ],
    "operator-linebreak": [
      2,
      "after"
    ],
    "space-infix-ops": 2,
    "space-unary-ops": 2,
    "no-implicit-coercion": [
      2,
      {
        "number": true,
        "boolean": true,
        "string": true
      }
    ],
    "no-with": 2,
    "no-multiple-empty-lines": 2,
    "linebreak-style": [
      2,
      "unix"
    ],
    "quotes": [
      2,
      "single"
    ],
    "no-mixed-spaces-and-tabs": 2,
    "no-trailing-spaces": 2,
    "comma-dangle": [
      2,
      "never"
    ],
    "brace-style": 2,
    "max-len": [
      2,
      120
    ],
    "consistent-this": [
      2,
      "$this"
    ],
    "yoda": [
      2,
      "never"
    ]
  }
}
```

ESLint version: `v1.3.1`

**Comment by maxlapides** [comment]
I have a related issue. Here's what my .eslintrc file looks like:

```
{
    "rules": {
        "indent" : [2, 2, {"SwitchCase": 1}],
    }
}
```

And then I get this error message:

```
Error: [project/path]/.eslintrc:
	Configuration for rule "indent" is invalid:
	Value "data["2"].SwitchCase" has additional properties.
  at validateRuleOptions ([project/path]/node_modules/eslint/lib/config-validator.js:88:15)
  at [project/path]/node_modules/eslint/lib/config-validator.js:101:13
  at Array.forEach (native)
  at Object.validate ([project/path]/node_modules/eslint/lib/config-validator.js:100:35)
  at loadConfig ([project/path]/node_modules/eslint/lib/config.js:99:19)
  at getLocalConfig ([project/path]/node_modules/eslint/lib/config.js:225:23)
  at Config.getConfig ([project/path]/node_modules/eslint/lib/config.js:363:22)
  at CLIEngine.getConfigForFile ([project/path]/node_modules/eslint/lib/cli-engine.js:382:29)
  at /Users/mlapides/.atom/packages/linter-eslint/lib/linter-eslint.coffee:90:29
  at exports.allowUnsafeNewFunction (/Users/mlapides/.atom/packages/linter-eslint/node_modules/loophole/lib/loophole.js:25:14)
  at Object.provider.lint (/Users/mlapides/.atom/packages/linter-eslint/lib/linter-eslint.coffee:89:11)
  at Promise.then._this.emitter.emit.linter (/Users/mlapides/.atom/packages/linter/lib/linter-registry.coffee:59:22)
  at LinterRegistry.triggerLinter (/Users/mlapides/.atom/packages/linter/lib/linter-registry.coffee:58:16)
  at /Users/mlapides/.atom/packages/linter/lib/linter-registry.coffee:52:17
  at Array.map (native)
  at /Users/mlapides/.atom/packages/linter/lib/linter-registry.coffee:50:27
```

(I replaced my actual project path with `[project/path]` in the above log.)

I'm posting this issue here because I'm not having the same issue when I run ESLint at the command line.

Thanks for any suggestions :D

**Comment by IanVS** [comment]
@maxlapides, my guess is that the local version of ESLint inside your project (which linter-eslint is using) is an older version which did not support the newer indent configuration that you're using.  If you don't have a version of ESLint inside your project, linter-eslint seems to use `0.24.1` which is quite old now. 

As for @nwhite89, yours befuddles me, as `indent` was added way back in `0.14`.  Do you get the same error when you run `node_modules/.bin/eslint .` from your project folder?

**Comment by maxlapides** [comment]
@IanVS ah yes, I did have an old version of ESLint installed in node_modules in my project. I didn't think to check on that because I had `Use Global Eslint` checked, but after playing with that checkbox a bit more I am realizing it doesn't seem to do anything haha

forcing an upgrade to the latest ESLint locally solved the problem! thanks!

**Comment by nwhite89** [comment]
I actually kept getting `Global ESlint` error a few times, it is actually working now :/ I removed my `node_modules` folder and ran `npm i` again and all seems to be working now may be a version issue also ?

**Comment by onlywei** [comment]
I still have this issue. My ESLint package is globally installed, and I don't want to install it locally since I use `gulp-eslint` instead. 

Global ESLint version is 1.3.1, and I do have the "Use Global Eslint" checkbox checked.

**Comment by IanVS** [comment]
What version of `gulp-eslint` are you using?  "Use Global ESLint" has been reported to not work correctly in a few cases (https://github.com/AtomLinter/linter-eslint/issues/191#issuecomment-137810088 & just above yours in https://github.com/AtomLinter/linter-eslint/issues/202#issuecomment-137316938).  Might be worth investigating if that is the culprit (and also opening an issue if that's truly the case).

**Comment by onlywei** [comment]
I am using `gulp-eslint` 1.0.0. I made a pull request that I think might address this, but am unsure if it's the ideal solution. Please take a look.

**Comment by IanVS** [comment]
@onlywei What is the error you are getting?  Have you tried `rm -rf node_modules && npm i` from within your project folder, to make sure you pull in the latest dependencies?  From what I can tell, the eslint from `node_modules/gulp-eslint/eslint` should be being used.

**Comment by onlywei** [comment]
@IanVS I am getting this error:

```
[Linter-ESLint] eslint binary not found locally, falling back to packaged one. Plugins won't be loaded and linting will possibly not work. (Try Use Global ESLint option, or install locally eslint to your project.)
```

Followed by:

```
Error: /Users/wei/development/urbancompass/src/js/.eslintrc: Configuration for rule "indent" is invalid: Value "data["2"].VariableDeclarator" has additional properties.
Error: /Users/wei/development/urbancompass/src/js/.eslintrc:
	Configuration for rule "indent" is invalid:
	Value "data["2"].VariableDeclarator" has additional properties.
  at validateRuleOptions (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config-validator.js:88:15)
  at /Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config-validator.js:101:13
  at Array.forEach (native)
  at Object.validate (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config-validator.js:100:35)
  at loadConfig (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:99:19)
  at getLocalConfig (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:225:23)
  at Config.getConfig (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:363:22)
  at CLIEngine.getConfigForFile (/Users/wei/.atom/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:382:29)
  at /Users/wei/.atom/packages/linter-eslint/lib/linter-eslint.coffee:90:29
  at exports.allowUnsafeNewFunction (/Users/wei/.atom/packages/linter-eslint/node_modules/loophole/lib/loophole.js:25:14)
  at Object.provider.lint (/Users/wei/.atom/packages/linter-eslint/lib/linter-eslint.coffee:89:11)
  at Promise.then._this.emitter.emit.linter (/Users/wei/.atom/packages/linter/lib/linter-registry.coffee:59:22)
  at LinterRegistry.triggerLinter (/Users/wei/.atom/packages/linter/lib/linter-registry.coffee:58:16)
  at /Users/wei/.atom/packages/linter/lib/linter-registry.coffee:52:17
  at Array.map (native)
  at /Users/wei/.atom/packages/linter/lib/linter-registry.coffee:50:27
```

I no longer get these errors after I modified the `linter-eslint.coffee` file in my config folder.

**Comment by steelbrain** [comment]
Fixed by v5.0.0
